lib/uklibid/libinfo.S

```diff
@@ -0,0 +1,55 @@
+#include <uk/config.h>
+#include <uk/libinfo.h>
+
+#define RECORD(type, data)				  \
+	.align 1					; \
+1:	.short type			/* type */	; \
+	.int   2f - 1b			/* len */	; \
+	data				/* data[] */	; \
+2:	.align 1
+
+.pushsection .uk_libinfo, "a"
+	.align 1
+3:	.int	4f - 3b			/* len */
+	.short	UKLI_LAYOUT		/* version */
+
+#if __GLOBALINFO__
+	/* Only one global header per unikernel is generated */
+	/* NOTE: Global metadata do not have a libname record */
+#if CONFIG_LIBUKLIBID_INFO_UKFULLVERSION
```

Comment:
There is the good chance that you also want to know the version of Unikraft a library is compiled with, in order to spot inconsistencies.

Reply:
This is a good point. I am going to add this as another config option.